packages/flutter_tools/bin/podhelper.rb

```diff
@@ -65,12 +65,34 @@ def flutter_additional_ios_build_settings(target)
   # Bundles are com.apple.product-type.bundle, frameworks are com.apple.product-type.framework.
   target_is_resource_bundle = target.respond_to?(:product_type) && target.product_type == 'com.apple.product-type.bundle'
 
+  # Detect pod-level EXCLUDED_ARCHS for the simulator written to Target Support Files xcconfig.
+  support_files_dir = File.join('Pods', 'Target Support Files', target.name)
+  if Dir.exist?(support_files_dir)
+    Dir.glob(File.join(support_files_dir, '*.xcconfig')).each do |xcfile|
+      File.foreach(xcfile) do |line|
+        if line =~ /^\s*EXCLUDED_ARCHS\b/ && line.include?('arm64')
+          abort "The pod `#{target.name}` config in #{xcfile} excludes the arm64 simulator architecture.\n" \
+            "This prevents running on arm64 simulators (iOS 26+). Remove `EXCLUDED_ARCHS[sdk=iphonesimulator*] = arm64` from the podspec/pod_target_xcconfig or update the pod to a version that supports arm64 simulators."
+        end
+      end
+    end
+  end
+
   target.build_configurations.each do |build_configuration|
     # Build both x86_64 and arm64 simulator archs for all dependencies. If a single plugin does not support arm64 simulators,
     # the app and all frameworks will fall back to x86_64. Unfortunately that case is not detectable in this script.
     # Therefore all pods must have a x86_64 slice available, or linking a x86_64 app will fail.
     build_configuration.build_settings['ONLY_ACTIVE_ARCH'] = 'NO' if build_configuration.type == :debug
 
+    # Xcode 16 and later require all simulator code to be built for arm64 on Apple Silicon Macs.
```

Comment:
Xcode 16 requires this too? I thought it was just Xcode 26 since Xcode 26 doesn't include the x86 simulator anymore?

Reply:
Per offline conversation:

Xcode 16 requires arm64 but supported x86 through rosetta. Xcode 26 no longer supports rosetta for simulators.

I think this comment is a little misleading, though, because it makes you think that you can't use `EXCLUDED_ARCHS` on Xcode 16 too. I would make the comment confined to Xcode 26. The historical context of Xcode 16 isn't super relevant. 

I'd just have the comment say something like "Xcode 26 no longer allows you to build x86-only architecture for the simulator."